### PR TITLE
refactor: handleNewBooking #3

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

Continuing refactor of handleNewBooking 
- https://github.com/calcom/cal.com/pull/15532
- https://github.com/calcom/cal.com/pull/15590

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x]  N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking/types.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import type { App } from "@prisma/client";
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/types.ts:1]: moved all the types here

File: packages/features/bookings/lib/handleNewBooking/createBooking.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    originalRescheduledBooking,
+    bookerEmail,
+    rescheduleReason,
+    eventType,
+    eventTypeId,
+    reqBodyUser,
+  });
+
+  return await saveBooking(newBookingData, originalRescheduledBooking, paymentAppData, organizerUser);
+}
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/createBooking.ts:99]: Refactored to smaller functions

File: packages/features/bookings/lib/handleNewBooking/ensureAvailableUsers.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    ) {
+      dateRangeForBooking = true;
+      break;
+    }
+  }
+
+  return dateRangeForBooking;
+};
+
+export async function ensureAvailableUsers(
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/ensureAvailableUsers.ts:50]: some refactoring and cleaning up

File: packages/features/bookings/lib/handleNewBooking/findBookingQuery.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import prisma from "@calcom/prisma";
+
+export const findBookingQuery = async (bookingId: number) => {
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/findBookingQuery.ts:3]: TODO: move these queries to repository in this PR https://github.com/calcom/cal.com/pull/15447

File: packages/features/bookings/lib/handleNewBooking/createBooking.ts [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+  smsReminderNumber: AwaitedBookingData["smsReminderNumber"];
+  eventTypeSlug: AwaitedBookingData["eventTypeSlug"];
+  organizerUser: CreateBookingParams["organizerUser"];
+  reqBodyRecurringEventId: ReqBodyWithEnd["recurringEventId"];
+  originalRescheduledBooking: OriginalRescheduledBooking | null;
+  bookerEmail: AwaitedBookingData["email"];
+  rescheduleReason: AwaitedBookingData["rescheduleReason"];
+  eventType: NewBookingEventType;
+  eventTypeId: EventTypeId;
+  reqBodyUser: ReqBodyWithEnd["user"];
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/createBooking.ts:182]: Possible Followup: I think we could structure it better. Data that goes directly to booking can be in `booking` prop and other things could be separate props like `reqBodyUser`, `isConfirmedByDefault` etc

Trying to minimize the props here and make this function more readable.

File: packages/features/bookings/lib/handleNewBooking/createBooking.ts [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  if (evt.team?.members) {
+    attendees.push(
+      ...evt.team.members.map((member) => ({
+        email: member.email,
+        name: member.name,
+        timeZone: member.timeZone,
+        locale: member.language.locale,
+      }))
+    );
+  }
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/createBooking.ts:162]: Followup: Simplification
```suggestion
 const teamMembers = event.team?.members ?? []
 return evt.attendees.concat(teamMembers).map(()=>{
 name: attendee.name,
    email: attendee.email,
    timeZone: attendee.timeZone,
    locale: attendee.language.locale,
 })
```

File: packages/features/bookings/lib/handleNewBooking/getRequiresConfirmationFlags.ts [hunk 7]
```diff
@@ -69,3 +69,5 @@ function determineIsConfirmedByDefault(
 ): boolean {
   return (!requiresConfirmation && price === 0) || userReschedulingIsOwner;
 }
+
+export type IsConfirmedByDefault = ReturnType<typeof getRequiresConfirmationFlags>["isConfirmedByDefault"];
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/getRequiresConfirmationFlags.ts:73]: It is just a boolean. So, we don't need to export it.

```suggestion
```

File: packages/features/bookings/lib/handleNewBooking/loadUsers.ts [hunk 8]
```diff
@@ -85,3 +85,5 @@ export const findUsersByUsername = async ({
     };
   });
 };
+
+export type AwaitedLoadUsers = Awaited<ReturnType<typeof loadUsers>>;
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/loadUsers.ts:89]: I think it's better to call it `LoadedUsers` or something similar.

Awaited doesn't make sense in name,

File: packages/features/bookings/lib/handleNewBooking/types.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+export type IsFixedAwareUser = User & {
+  isFixed: boolean;
+  credentials: CredentialPayload[];
+  organization: { slug: string };
+  priority?: number;
+};
+
+export type NewBookingEventType = AwaitedGetDefaultEvent | getEventTypeResponse;
+
+export type IsConfirmedByDefault = ReturnType<typeof getRequiresConfirmationFlags>["isConfirmedByDefault"];
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/types.ts:64]:
```suggestion
```

File: packages/lib/defaultEvents.ts [hunk 10]
```diff
@@ -168,3 +168,5 @@ export const getUsernameList = (users: string | string[] | undefined): string[]
 };
 
 export default defaultEvents;
+
+export type AwaitedGetDefaultEvent = Awaited<ReturnType<typeof getDefaultEvent>>;
```
Comment (hariombalhara) [inline packages/lib/defaultEvents.ts:172]:
```suggestion
export type DefaultEvent = Awaited<ReturnType<typeof getDefaultEvent>>;
```

**Review by hariombalhara — APPROVED** [review]

**Comment by Udit-takkar** [comment]
False alarm by sentry. /v1/booking endpoint is working fine. Few bad requests were made to create booking

**Pull request merged**